### PR TITLE
added support for s3 files

### PR DESCRIPTION
**Pull request by Anu-Ra-g**
This PR adds the `remote_options` parameter in necessary functions to be compatible with S3.

### CODE REVIEW

**Comment by Anu-Ra-g** [comment]
@martindurant It was in my local but I didn't push in the original mapping branch.

File: kerchunk/grib2.py [hunk 1]
```diff
@@ -811,7 +811,10 @@ def extract_datatree_chunk_index(
 
 
 def _map_grib_file_by_group(
-    fname: str, mapper: Optional[Callable] = None, storage_options=None
+    fname: str,
+    mapper: Optional[Callable] = None,
+    storage_options=None,
+    remote_options: Dict = None,
```
Comment (martindurant) [inline kerchunk/grib2.py:817]: Why is storage_options and remote_options different? We are only reading one remote file, no?

Aside: the type annotation should be `Optional[dict]` or `dict | None`.

Comment (Anu-Ra-g) [reply kerchunk/grib2.py:817]: It is for the remote_options parameter in `grib_tree`
```
def grib_tree(
    message_groups: Iterable[Dict],
    remote_options=None,
) -> Dict:
```

Comment (martindurant) [reply kerchunk/grib2.py:817]: But why is it different

Comment (Anu-Ra-g) [reply kerchunk/grib2.py:817]: Does `remote_options` and `target_options` come under the `storage_options`?

Comment (martindurant) [reply kerchunk/grib2.py:817]: I don't understand your question. You are trying to add both storage_options and remote_options to this function signature, and I don't see why they would be different - both reference the same remote grib file(s), no?

Comment (Anu-Ra-g) [reply kerchunk/grib2.py:817]: I though it'd be confusing. So I made them different.  I'll make it `storage_options` across the code.

Comment (Anu-Ra-g) [reply kerchunk/grib2.py:817]: I got confused with the signature of `grib_tree` function,
```
def grib_tree(
    message_groups: Iterable[Dict],
    remote_options=None,
   ```

Comment (Anu-Ra-g) [reply kerchunk/grib2.py]: @martindurant  I've made the updates.

**Pull request merged**